stytch/api/base.py

```diff
@@ -1,10 +1,15 @@
 import json
+from typing import TYPE_CHECKING, Any, Dict, Optional, Set
+
 import requests
+from requests.auth import HTTPBasicAuth
+
+from stytch.api.decorator import throw_stytch_exception
 from stytch.version import __version__
 
-from typing import Dict, Set
+if TYPE_CHECKING:
+    from stytch.client import Client
```

Comment:
Oooh, TIL `typing.TYPE_CHECKING`. That's kinda nice actually 🙂

Reply:
I have some more ideas to clean this up to avoid the really ugly circular dependencies. I can also look into adding more GH actions to test on older versions. Let me at minimum ensure this builds on 3.7 since that was a problem before.

Reply:
<img width="761" alt="image" src="https://user-images.githubusercontent.com/119902778/206373209-e703b2ff-caa5-4b0c-8ef8-b8ea3656541d.png">
Alright well this is just unfortunate.

Reply:
Oof. Well it turns out I still have a Python 3.7 around, and it at least imports fine!
```
$ python
Python 3.7.13 (default, Sep  1 2022, 13:47:02)
[Clang 13.1.6 (clang-1316.0.21.2.5)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import stytch
>>> import inspect
>>> inspect.getsource(stytch)
'from stytch.client import Client\nfrom stytch.version import __version__\n\n__all__ = ["Client", "__version__"]\n'
```
My test is that it your new imports (`stytch.*` instead of `.*`).

Let me know if you want me to check anything specific (or just find me on Slack and I can let you drive).